jsonrpc-specs/src/methods/discv5.json

```diff
@@ -0,0 +1,567 @@
+					},
+					"recipientPort": {
+						"description": "recipient UDP port",
+						"type": "number"
+					}
+				}
+			}
+		}
+	},
+	{
+		"name": "discv5_findNodes",
+		"summary": "Send a FINDNODES request for nodes that fall within the given set of distances, to the designated peer and wait for a response.",
+		"params": [
+			{
+				"name": "node_id",
+				"required": true,
+				"schema": {
+					"title": "Ethereum node record",
+					"$ref": "#/components/schemas/Enr"
+				}
+			},
+			{
+				"name": "distances",
+				"required": true,
+				"schema": {
+					"title": "distance",
+					"type": "array",
+					"items": {
+						"type": "number"
+					}
+				}
+			}
+		],
+		"result": {
+			"name": "findNodesResult",
+			"description": "Returns nodes in a given distance",
+			"schema": {
+				"title": "FOUNDNODES message",
+				"type": "array",
+				"items": { "$ref": "#/components/schemas/Enr" }
+			}
+		}
+	},
+	{
+		"name": "discv5_talk",
+		"summary": "Send a TALKREQUEST request with a payload to a given peer and wait for response.",
+		"params": [
+			{
+				"name": "node_id",
+				"required": true,
+				"schema": {
+					"title": "Ethereum node record",
+					"$ref": "#/components/schemas/Enr"
+				}
+			},
+			{
+				"name": "protocol",
+				"required": true,
+				"description": "protocol element of message-data",
+				"schema": {
+					"title": "protocol id",
+					"$ref": "#/components/schemas/hexString"
+				}
+			},
+			{
+				"name": "payload",
+				"required": true,
+				"description": "request element of message-data",
+				"schema": {
+					"$ref": "#/components/schemas/hexString"
+				}
+			}
+		],
+		"result": {
+			"name": "talkResult",
+			"description": "Returns TALKRESPONSE message as hex string",
+			"schema": {
+				"$ref": "#/components/schemas/hexString"
+			}
+		}
+	},
+	{
+		"name": "discv5_recursiveFindNodes",
+		"summary": "Lookup a target node within in the network",
+		"params": [
+			{
+				"name": "node_id",
+				"required": true,
+				"schema": {
+					"title": "Ethereum node record",
```

Comment:
I think here it actually makes most sense to provide the node id. As it is your "target" and you are requesting it because you don't have the ENR.